x-pack/plugins/security_solution/server/lib/entity_analytics/asset_criticality/asset_criticality_service.ts

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { isEmpty } from 'lodash/fp';
+import type { ExperimentalFeatures } from '../../../../common';
+import type { AssetCriticalityRecord } from '../../../../common/api/entity_analytics';
+import type { AssetCriticalityDataClient } from './asset_criticality_data_client';
+
+interface CriticalityIdentifier {
+  id_field: string;
+  id_value: string;
+}
+
+interface IdentifierValuesByField {
+  [idField: string]: string[];
+}
+
+export interface AssetCriticalityService {
+  getCriticalitiesByIdentifiers: (
+    identifiers: CriticalityIdentifier[]
+  ) => Promise<AssetCriticalityRecord[]>;
+  isEnabled: () => boolean;
+}
+
+const isCriticalityIdentifierValid = (identifier: CriticalityIdentifier): boolean =>
+  !isEmpty(identifier.id_field) && !isEmpty(identifier.id_value);
+
+const groupIdentifierValuesByField = (
+  identifiers: CriticalityIdentifier[]
+): IdentifierValuesByField =>
+  identifiers.reduce((acc, id) => {
+    if (!acc[id.id_field]) {
```

Comment:
probably can be replaced by 

``acc[id.id_field] ??= []``

Reply:
4db7a1777dd76c6b4089c48ee0162297e78ae398